docs/source/app-dev/daml-lf-translation.rst

```diff
@@ -245,3 +245,29 @@ These translate to the DAML-LF record declarations:
 
 	record DoNothing ↦ {}
 	record Transfer ↦ { newOwner: Party }
+
+Names with special characters
+*****************************
+
+All names in DAML—of types, templates, choices, fields, and variant data constructors—are translated to the more restrictive rules of DAML-LF.  ASCII letters, digits, and ``_`` underscore are unchanged in DAML-LF; all other characters must be mangled in some way, as follows:
+
+- ``$`` changes to ``$$``,
+- Unicode codepoints less than 65536 translate to ``$uABCD``, where ``ABCD`` are exactly four (zero-padded) hexadecimal digits of the codepoint in question, using only lowercase ``a-f``, and
+- Unicode codepoints greater translate to ``$UABCD1234``, where ``ABCD1234`` are exactly eight (zero-padded) hexadecimal digits of the codepoint in question, with the same ``a-f`` rule.
+
+.. list-table::
+   :widths: 10 15
+   :header-rows: 1
+
+   * - DAML name
+     - DAML-LF identifier
+   * - ``Foo_bar``
+     - ``Foo_bar``
+   * - ``baz'``
+     - ``baz$u0027``
+   * - ``:+:``
+     - ``$u003a$u002b$u003a``
+   * - ``naïveté``
+     - ``na$u00efvet$u00e9``
+   * - ``:🙂:``
```

Comment:
Is that a valid DAML identifier?

Reply:
Not yet. :trollface:

Reply:
But DAML does allow for Unicode in identifiers.

Reply:
right but only things that are letters afaik.

Reply:
No, you can use emojis as long as they are single code-point. Eg this is a valid type:

```
data Wut a = (:🙂:) with x : a; y : a
           | (:🙁:) with x : a; y : a
```

Reply:
> But i guess functions and variant constructors are not identifiers?

Function and variant constructor names are identifiers as well. So it's good we have documentation for this as well.

Reply:
FWIW I checked every example here in VS Code, some with `TypeOperators` on; all were accepted.